src/executables/server.c

```diff
@@ -403,26 +406,32 @@ main (int argc, char **argv)
   {				/* to make indent happy */
     if (argc < 2)
       {
-        PRINT_AND_LOG_ERR_MSG ("Usage: server databasename\n");
-        return 1;
+	PRINT_AND_LOG_ERR_MSG ("Usage: server databasename\n");
+	return 1;
       }
 
     if (er_init (NULL, ER_NEVER_EXIT) != NO_ERROR)
       {
-        PRINT_AND_LOG_ERR_MSG ("Failed to initialize error manager\n");
-        return 1;
+	PRINT_AND_LOG_ERR_MSG ("Failed to initialize error manager\n");
+	return 1;
       }
     thread_initialize_manager (thread_p);
     fprintf (stdout, "\nThis may take a long time depending on the amount " "of recovery works to do.\n");
-
+    // *INDENT-OFF*
+    scope_exit <std::function<void (void)>> print_on_exit ([thread_p] ()
+    {
+      PRINT_AND_LOG_ERR_MSG ("%s\n", er_msg ());
+      fflush (stderr);
+    });
```

Comment:
- Error should be printed only if there was an error, reflected by ret_val's value.
- No point on capturing `thread_p`, but a reference to `ret_val` has to be captured.
```suggestion
    scope_exit <std::function<void (void)>> print_err_on_exit ([&ret_val] ()
    {
      if (ret_val != 0)
        {
          PRINT_AND_LOG_ERR_MSG ("%s\n", er_msg ());
          fflush (stderr);
        }
    });
```